packages/valory/skills/twitter_scoring_abci/rounds.py

```diff
@@ -89,6 +89,28 @@ def latest_mention_tweet_id(self) -> dict:
         return cast(dict, self.db.get_strict("latest_mention_tweet_id"))
 
 
+class OpenAICallCheckRound(CollectSameUntilThresholdRound):
+    """OpenAICallCheckRound"""
+
+    payload_class = TwitterCollectionPayload
+    synchronized_data_class = SynchronizedData
+
+    API_CALL_EXCEEDED = "API_CALLS_EXCEEDED"
+    API_CALL_REMAINING = "API_CALL_REMAINING"
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            if self.most_voted_payload == self.API_CALL_EXCEEDED:
+                return self.synchronized_data, Event.API_ERROR
+            return self.synchronized_data, Event.DONE
+        if not self.is_majority_possible(
+            self.collection, self.synchronized_data.nb_participants
+        ):
+            return self.synchronized_data, Event.NO_MAJORITY
+        return None
```

Comment:
All this is unnecessary if you use the `none_event` class attribute. See https://github.com/valory-xyz/IEKit/pull/78#discussion_r1305225837.